### PR TITLE
test: add tests for CheckoutDocument

### LINKED ISSUE #198: CheckoutDocument tests

### PR DESCRIPTION
**Pull request by kaitoz11**

### CODE REVIEW

File: tests/Application.Tests.Integration/Borrows/Commands/CheckoutDocumentTest.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+namespace Application.Tests.Integration.Borrows.Commands;
+
+public class CheckoutDocumentTest : BaseClassFixture
+{
+    public CheckoutDocumentTest(CustomApiFactory apiFactory) : base(apiFactory)
+    {
+    }
+
+    [Fact]
+    public async Task ShouldCheckoutDocument_WhenBorrowRequestDocument()
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Borrows/Commands/CheckoutDocumentTest.cs]: The When phrase is not that meaningful

File: tests/Application.Tests.Integration/Borrows/Commands/CheckoutDocumentTest.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            .WithMessage("Document is not available.");
+
+        // Cleanup
+        Remove(borrow);
+        Remove(document);
+        Remove(user);
+    }
+
+    [Fact]
+    public async Task ShouldThrowConflictException_WhenBorrowRequestStatusIsApproved()
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Borrows/Commands/CheckoutDocumentTest.cs]: The When phrase is wrong

**Review by ChienNQuang — COMMENTED** [review]
This test does not cover the case which the checkout date is not the borrow date, I shall add this later

File: tests/Application.Tests.Integration/Borrows/Commands/CheckoutDocumentTest.cs (hunk 1 above)
Comment (kaitoz11) [reply tests/Application.Tests.Integration/Borrows/Commands/CheckoutDocumentTest.cs]: fixed at https://github.com/ProFile-org/ProFile-BE/pull/213/commits/401961fa40831d6b32fd702618aa42b26465614a

File: tests/Application.Tests.Integration/Borrows/Commands/CheckoutDocumentTest.cs (hunk 2 above)
Comment (kaitoz11) [reply tests/Application.Tests.Integration/Borrows/Commands/CheckoutDocumentTest.cs]: fixed at https://github.com/ProFile-org/ProFile-BE/pull/213/commits/401961fa40831d6b32fd702618aa42b26465614a

**Comment by kaitoz11** [comment]
> This test does not cover the case which the checkout date is not the borrow date, I shall add this later

Do I need to update the `borrowRequest.BorrowTime` in checkout flow?

**Pull request closed without merging**